src/daoreg.cpp

```diff
@@ -442,19 +447,22 @@ void daoreg::createselloffer (
   auto by_offer_match = offer_t.get_index<eosio::name("byoffermatch")>();
 
   // offer match
+
   auto boitr_sell = by_offer_match.lower_bound( 
       ( uint128_t(0xF & util::type_buy_offer) << 124 ) 
      + ( uint128_t(0xF & util::status_active) << 122 )
      + ( uint128_t(0xF & token_id) << 120 )
      + ( uint128_t(0xFFFFFFFFFFFFFFFF & price_per_unit.amount ) << 56 )  
     );
+  
+  bool offer_not_exists = (boitr_sell == by_offer_match.end() || boitr_sell -> type != util::type_buy_offer);
```

Comment:
this is not correct, it should be an AND

Reply:
Done.